### DISCUSSION TITLE
PWs Charging from Grid after update to 24.12.3

### DISCUSSION THREAD

**Discussion by youzer-name** [jasonacox/Powerwall-Dashboard#486] [General]
My Powerwalls updated last night from 24.4.0 to 24.12.3. A little later, one of my cars started its scheduled charge. I have a home automation routine that sets the reserve to 100% anytime a car is charging to prevent charging the car from the PW. This has always worked as expected, with the batteries holding at whatever state they were at when the reserve was switched. However this morning the batteries started to charge from the grid when the reserve was set to 100%.

![image](https://github.com/jasonacox/Powerwall-Dashboard/assets/57046429/772703c2-3a08-438e-86a5-fa5b72849ba5)

The white shading indicates the car was charging. The orange shows that the batteries were charging from the grid. 

The recently-restored alerts panel showed this:

![image](https://github.com/jasonacox/Powerwall-Dashboard/assets/57046429/9e4c28e9-cafa-45fe-8a25-8b04b50a5ed1)

The first line shows "ChargeOnlyFromSolarOverride" 

In the Tesla app, grid charging is set to "No".

To rule out any weirdness with the API that the home automation system is using, I tried setting the reserve to 100% directly in the Tesla app. That also initiated grid charging.

I'm going to put a ticket in with Tesla Energy support to see if they can figure out what is going on. If anyone else is on 24.12.3, are you seeing the same thing when you increase the reserve setting? 

My setup is a GW2 with 3x Powerall 2's.

**Comment by BJReplay** [comment]
> I have a home automation routine that sets the reserve to 100% anytime a car is charging to prevent charging the car from the PW.

I would have thought that the best approach would be to set your reserve to the current State of Charge.  That should cause the PWs to hold charge - regardless of the grid charging setting.

> In the Tesla app, grid charging is set to "No".

My guess - only that, because I'm on a much older version of firmware, and don't have an EV - is that the latest firmware isn't obeying the don't charge from grid setting.

**Reply by jasonacox** [reply]
Interesting. Thanks for sharing @youzer-name !

> set your reserve to the current State of Charge

I agree with @BJReplay - I have my scripts set the reserve to "current charge level" when I want to have the batteries stop charge/discharge at night (it will still charge via solar during the day). But I also have grid charging set, so your case doesn't make sense.

```bash
python3 -m pypowerwall set -current
```

On Firmware 24.12.3 - Have you noticed anything new? On a positive note, it is good to see the new vitals alert metrics are still working. :)

**Reply by youzer-name** [reply]
Yes, setting to the current charge level instead of 100% would fix the issue. 

Tesla support's answer was to completely turn off 'charge from grid', changing it from user preference to a hard off. They blamed the 'optimcaster', saying it was charging based on lowest cost. The thing is, I also don't have TOU rates set up or available to me from my electric company. I have 1:1 net metering, so a kWh used = a kWh returned to the grid. In any case, if there's a toggle to say 'don't charge from grid', then it shouldn't charge from grid' when that toggle is off.

**Reply by BJReplay** [reply]
> I have 1:1 net metering, so a kWh used = a kWh returned to the grid.

Except for round trip losses, so a kWh used from the grid equals around 870Wh returned to the grid :)
